### ISSUE TITLE
Always provide default values for properties

### ISSUE THREAD

**Issue by schneiderpat** [jburzynski/TypeGen#34]
Hey :)
Is there an option to always generate default values in the generated typescript class without annotating every property with `[TSDefaultValue]`?

Greets Patrick

**Comment by jburzynski** [comment]
Hi! Can you give an example? Do you mean something like a map of `c# type -> ts type` pairs (to specify which cs type corresponds to which ts type)?

**Comment by jburzynski** [comment]
Hi! Can you give an example? Do you mean providing a "global" default value for all properties?

**Comment by schneiderpat** [comment]
Currently we have to to this:
```csharp
[TSDefaultValue("")]
public string FirstName { get; set; }

[TSDefaultValue("")]
public string LastName { get; set; }
```

We need to generate the classes with default values because we have another generator to create formgroups in an angular in a special way.
A global Setting to generate default values for specific types would be awesome.

**Comment by jburzynski** [comment]
This seems like a useful feature for some cases. I'll aim to add it in the next release (the next release will also have the fix to #33).

I'll keep you updated with any progress in this topic. Cheers!

**Comment by schneiderpat** [comment]
Thanks a lot :)

Greets!

**Comment by schneiderpat** [comment]
How can I install the nuget-dotnetcli tool to test it?

**Comment by schneiderpat** [comment]
An option with custom default mappings would solve the problem and would be very flexiable.
```ts
// tgconfig.json
{
    "emptyValueMappings: {
        "Date | null": "null",
        "number": "-1"
    }
}
```
And in your `EmptyValue.cs` you check for that option and iterate through them first. By checking them first you add the possibility to override your defaults.

**Comment by jburzynski** [comment]
You can download the `TypeGen.DotNetCli` package from NuGet and then specify custom source when installing the tool:

```
dotnet tool install TypeGen.DotNetCli -g --add-source "PACKAGE_FOLDER_PATH"
```

I've dropped support for per-project tools, because they are going to be abandoned by Microsoft anyway (I gave a more detailed 'rationale' for dropping per-project tools in [this post](https://github.com/jburzynski/TypeGen/issues/33#issuecomment-441389823)), so right now TypeGen.DotNetCli is only a global tool.

About empty values, I'm thinking about removing the option that exists right now (`generateEmptyValues`) and replacing it with the option you suggested. I think there is no need for an additional option, if there's already an option to specify values for types. Do you think this would be ok?

**Comment by schneiderpat** [comment]
You are right, an additional option is not necessary. 
Removing `generateEmptyValues` is ok, when we can specify default values ourselves. 

Thanks again for your work!

**Comment by jburzynski** [comment]
You're welcome! Thanks!

I'll let you know when it's implemented.

**Comment by jburzynski** [comment]
There is a new version on the dev branch (nuget package). The option is named `defaultValuesForTypes` and you would use it like this (the same as in your post):

```json
"defaultValuesForTypes": {
    "number": "-1",
    "Date | null": "null"
}
```

It would be cool if you could check again if it works! Thanks! :)

**Comment by schneiderpat** [comment]
I tested and it works awesome!
I´m  waiting for the official release ;)

Thanks again for your great work!

**Comment by jburzynski** [comment]
No problem! Thanks for the info!

I've released version 2.0.1 (major version change is because of the changed syntax in the `TypeGen` nuget package - it's now the same as in the .NET CLI tool) and it contains the new functionality (usage is exactly the same). Hope it works! :)
